src/xo-app/settings/remotes/index.js

```diff
@@ -31,196 +32,210 @@ const remoteTypes = {
+          placeholder={userData.placeholder}
+          value={remote.host}
+        />
+        :
+        <Text
+          onChange={v => _changeUrlElement(v, 'path')}
+          placeholder={userData.placeholder}
+          value={remote.path}
+        />
+      </span>
+    },
+    name: _('remoteDevice')
   }
-}
-
-@injectIntl
-class SmbRemote extends AbstractRemote {
-  _renderRemoteInfo () {
-    const { remote } = this.props
-    return <span>
-      <strong className='text-info'>\\</strong>
-      <Text value={remote.host} onChange={v => this._changeUrlElement(v, 'host')} />
-      <strong className='text-info'>\</strong>
-      <span>
-        <Text value={remote.path} onChange={v => this._changeUrlElement(v, 'path')} placeholder={this.props.intl.formatMessage(messages.remoteSmbPlaceHolderRemotePath)} />
+]
+const COLUMNS_SMB_REMOTE = [
+  {
+    itemRenderer: () => '',
+    name: _('remoteTypeSmb')
+  },
+  {
+    itemRenderer: (remote, userData) => {
+      const _changeUrlElement = (value, element) => {
+        remote[element] = value
+        const url = format(remote)
+        return editRemote(remote, {url})
+      }
+      return <span>
+        <strong className='text-info'>\\</strong>
+        <Text value={remote.host} onChange={v => _changeUrlElement(v, 'host')} />
+        <strong className='text-info'>\</strong>
+        <span>
+          <Text
+            onChange={v => _changeUrlElement(v, 'path')}
+            placeholder={userData.placeholder}
+            value={remote.path}
+          />
+        </span>
+      </span>
+    },
+    name: _('remoteShare')
+  },
+  {
+    itemRenderer: (remote, userData) => {
+      const _changeUrlElement = (value, element) => {
+        remote[element] = value
+        const url = format(remote)
+        return editRemote(remote, {url})
+      }
+      return <span>
+        <Text value={remote.username} onChange={v => _changeUrlElement(v, 'username')} />
+        :
+        <Password value='' onChange={v => _changeUrlElement(v, 'password')} placeholder={userData.placeholder} />
+        @
+        <Text value={remote.domain} onChange={v => _changeUrlElement(v, 'domain')} />
       </span>
-    </span>
+    },
+    name: _('remoteAuth')
   }
+]
+COLUMNS_LOCAL_REMOTE.splice(1, 0, COLUMN_NAME)
+COLUMNS_LOCAL_REMOTE.splice(3, 0, COLUMN_STATE)
 
-  _renderAuthInfo () {
-    const { remote } = this.props
-    return <span>
-      <Text value={remote.username} onChange={v => this._changeUrlElement(v, 'username')} />
-      :
-      <Password value='' onChange={v => this._changeUrlElement(v, 'password')} placeholder={this.props.intl.formatMessage(messages.remotePlaceHolderPassword)} />
-      @
-      <Text value={remote.domain} onChange={v => this._changeUrlElement(v, 'domain')} />
-    </span>
-  }
+COLUMNS_NFS_REMOTE.splice(1, 0, COLUMN_NAME)
+COLUMNS_NFS_REMOTE.splice(3, 0, COLUMN_STATE)
+
+COLUMNS_SMB_REMOTE.splice(1, 0, COLUMN_NAME)
+COLUMNS_SMB_REMOTE.splice(3, 0, COLUMN_STATE)
```

Comment:
Why not adding them directly in the arrays?

Reply:
Because I don't want to have same multiple lines of code repeated in multiple times.

Reply:
I get that but you could keep `COLUMN_STATE` and `COLUMN_NAME` as they are and create the `COLUMNS` arrays using them:
```js
const COLUMNS1 = [
  COLUMN_NAME,
  {
    name: '...',
    itemRenderer: () => {}
  }
]

const COLUMNS2 = [
  COLUMN_NAME,
  {
    name: '...',
    itemRenderer: () => {}
  },
  COLUMN_STATE
]
```
I think it would be simpler and clearer to someone who reads the code. What do you think?